src/stream/src/executor/source/source_executor.rs

```diff
@@ -561,7 +513,7 @@ impl<S: StateStore> SourceExecutor<S> {
                             }
                             Mutation::Throttle(actor_to_apply) => {
                                 if let Some(throttle) = actor_to_apply.get(&self.actor_ctx.id) {
-                                    self.source_ctrl_opts.rate_limit = *throttle;
+                                    self.rate_limit_rps = *throttle;
```

Comment:
please also add a check if the rate limit really changes

Reply:
Fixed.